sdk/client/src/move_deserialize.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Error, Result};
+use diem_json_rpc_types::views::{AccountStateWithProofView, EventWithProofView};
+use diem_types::{
+    account_state::AccountState,
+    account_state_blob::AccountStateBlob,
+    contract_event::{ContractEvent, EventWithProof},
+};
+use move_core_types::{language_storage::TypeTag, move_resource::MoveResource};
+use serde::de::DeserializeOwned;
+use std::convert::TryFrom;
+
+/// Wrapper for a deserialized Move event and its containing `ContractEvent`
+#[derive(Debug, Clone)]
+pub struct Event<T: MoveResource + DeserializeOwned> {
+    /// The deserialized event type
+    data: T,
+    event: ContractEvent,
+}
+
+impl<T: MoveResource + DeserializeOwned> Event<T> {
+    pub fn data(&self) -> &T {
+        &self.data
+    }
+
+    pub fn event(&self) -> &ContractEvent {
+        &self.event
+    }
+}
+
+/// Deserialize and return the Move events of type `T` in `events`
+/// The type `T` must match the specified event types in `events`
+pub fn get_events<T: MoveResource + DeserializeOwned>(
+    events: Vec<EventWithProofView>,
+) -> Result<Vec<Event<T>>> {
+    let events_with_proof: Vec<EventWithProof> = events
+        .into_iter()
+        .map(|e| {
+            bcs::from_bytes::<EventWithProof>(e.event_with_proof.inner()).map_err(Error::decode)
+        })
+        .collect::<Result<Vec<EventWithProof>>>()?;
+    let event_type_tag = TypeTag::Struct(T::struct_tag());
+    events_with_proof
+        .into_iter()
+        .map(|e| {
+            // Check that `T` matches the type specified in `e`
```

Comment:
@bmwill Making sure this looks ok to you (particularly constructing my own `Error::decode` with a long message)